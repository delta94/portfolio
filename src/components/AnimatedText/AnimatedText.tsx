import React from 'react';
import { createUseStyles } from 'react-jss';
import TextLoop from 'react-text-loop';
import { CustomTheme } from '../../theme';

interface AnimatedTextProps {}

const useStyles = createUseStyles((theme: CustomTheme) => ({
  textColor: {
    color: theme.colors.white,
  },
  animatedText: {
    marginLeft: theme.spacing(1),
  },
}));

const AnimatedText: React.FC<AnimatedTextProps> = () => {
  const classes: Record<'textColor' | 'animatedText', string> = useStyles();

  return (
    <div className={classes.textColor}>
      I am a
      <span className={classes.animatedText}>
        <TextLoop
          springConfig={{ stiffness: 180, damping: 8 }}
          interval={1500}
          children={['Front End Developer', 'Photography Lover']}
        />
      </span>
    </div>
  );
};

export { AnimatedText };
