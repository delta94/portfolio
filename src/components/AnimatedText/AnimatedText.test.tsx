import { render } from '@testing-library/react';
import React from 'react';
import { TestContextProvider } from '../../TestContextProvider';
import { AnimatedText } from './AnimatedText';

test('AnimatedText renders correctly', () => {
  const { asFragment } = render(
    <TestContextProvider>
      <AnimatedText />
    </TestContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
