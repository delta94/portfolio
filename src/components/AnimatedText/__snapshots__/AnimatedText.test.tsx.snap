// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedText renders correctly 1`] = `
<DocumentFragment>
  <div
    class="textColor-0-2-1"
  >
    I am a
    <span
      class="animatedText-0-2-2"
    >
      <div
        class="x0 x1 x2 "
      >
        <div
          style="transition: width 150ms linear; height: auto; width: auto;"
        >
          <div
            class="x0 x3 x4 x5"
            style="opacity: 1; transform: translateY(0px); position: relative;"
          >
            Front End Developer
          </div>
        </div>
      </div>
    </span>
  </div>
</DocumentFragment>
`;
